packages/medusa/src/api/admin/products/route.ts

```diff
@@ -30,11 +30,17 @@ export const GET = async (
 }
 
 export const POST = async (
-  req: AuthenticatedMedusaRequest<HttpTypes.AdminCreateProduct>,
+  req: AuthenticatedMedusaRequest<
+    HttpTypes.AdminCreateProduct & {
+      additional_data?: Record<string, unknown>
+    }
```

Comment:
nit(non-blocking): We might want to create a dedicated type instead of inline at some point - wdyt?

Reply:
Sure. Let me do that now, instead of leaving it to later